iron-input.html

```diff
@@ -103,6 +103,42 @@
       'keypress': '_onKeypress'
     },
 
+    registered: function() {
+      // Feature detect whether we need to patch dispatchEvent (i.e. on FF and IE).
+      var input = document.createElement('input');
+      if (!this._canDispatchEventOnDisabled(input)) {
+        this._origDispatchEvent = this.dispatchEvent;
+        this.dispatchEvent = this._dispatchEventFirefoxIE;
+      }
+    },
+
+    _canDispatchEventOnDisabled: function(input) {
```

Comment:
Since this feature detection is for this element only, it can probably just create an input internally instead of receiving an input argument.